File: src/main/java/com/sdl/selenium/web/SearchText.java
package com.sdl.selenium.web;

import java.util.*;

public class SearchText {

    private String value;
    private List<SearchType> searchTypes = new ArrayList<>();

    public SearchText(String value, List<SearchType> searchTypes) {
        this.value = value;
        this.searchTypes = searchTypes;
    }

    public SearchText(String value, SearchType... searchTypes) {
        this.value = value;
        setSearchTypes(searchTypes);
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public List<SearchType> getSearchTypes() {
        return searchTypes;
    }

    public void setSearchTypes(SearchType... searchTypes) {
        if (searchTypes != null) {
            if (searchTypes.length == 0) {
                Collections.addAll(this.searchTypes, SearchType.EQUALS);
            } else {
                Collections.addAll(this.searchTypes, searchTypes);
            }
        } else {
            this.searchTypes.addAll(new XPathBuilder().defaultSearchTextType);
        }
        new XPathBuilder().cleanUpSearchType(this.searchTypes);
    }
}

File: dotnet/TestyForC/Web/SearchText.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TestyForC.Web;

namespace Testy.Web
{
    public class SearchText
    {
        private String value;
        private List<SearchType> searchTypes = new List<SearchType>();

        public SearchText(String value, List<SearchType> searchTypes)
        {
            this.Value = value;
            this.SearchTypes = searchTypes;
        }

        public string Value { get => value; set => this.value = value; }
        public List<SearchType> SearchTypes { get => searchTypes; set => searchTypes = value; }
    }
}
